### PR TITLE
Add containerd package and cpu,memory,blkio data streams

### PR DESCRIPTION
**Pull request by MichaelKatsoulis**
## What does this PR do?

After merging https://github.com/elastic/beats/pull/29247 which introduced `containerd` module this PR
creates containerd package with cpu, memory and blkio data streams.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

1. Bring up the elastic-stack version 8.1.0-SNAPSHOT
`elastic-package up  --version=8.1.0-SNAPSHOT -v -d `
2. Bring up a Kubernetes cluster with kind and containerd runtime `kind create cluster --image 'kindest/node:v1.21.1`
3. `Docker exec` in the kind kubernetes node and configure `containerd` to server metrics. Follow instructions [here](https://containerd.io/docs/getting-started/).
4. Deploy elastic agent with the same version and enrol it to fleet. 
5. Add containerd integration in a policy and assign it to the agent.
6. Watch containerd cpu, memory and blkio metrics being populated.

## Related issues

- Closes https://github.com/elastic/integrations/issues/2373

## Screenshots

containerd_integration

containerd_readme

containerd_configuration

containerd_data_streams

discover_cpu

discover_memory

discover_blkio

containerd_dash

### CODE REVIEW

File: packages/containerd/data_stream/blkio/fields/fields.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        Bytes read during the life of the container
+
+    - name: write.ops
+      type: long
+      description: >
+        Number of writes during the life of the container
+
+    - name: write.bytes
+      type: long
+      format: bytes
```
Comment (tetianakravchenko) [inline packages/containerd/data_stream/blkio/fields/fields.yml:26]: should `units` and `metric_type` be added as well?

for other fields as well - https://github.com/elastic/observability-dev/blob/fef856060729b1ff233774941711303a59a77b30/docs/dc/integrations/development/generic_guidelines.md#document-all-fields

File: packages/containerd/data_stream/blkio/fields/ecs.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+- external: ecs
+  name: container.id
```
Comment (tetianakravchenko) [inline packages/containerd/data_stream/blkio/fields/ecs.yml:2]: should `dimension: true` be set for this fields? as I see usually it is set on *.name field, but it is not possible to expose name, right?

the same for memory and cpu datastreams

File: packages/containerd/data_stream/blkio/fields/ecs.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+- external: ecs
```
Comment (tetianakravchenko) [inline packages/containerd/data_stream/blkio/fields/ecs.yml:1]: in sample event there is an ecs.version, should it be added here?

actually I thought that those fields are more or less a default for each integration - https://github.com/elastic/integrations/blob/master/packages/kubernetes/data_stream/container/fields/ecs.yml, no?

File: packages/containerd/data_stream/cpu/fields/fields.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      type: scaled_float
+      format: percent
+      unit: percent
+      metric_type: gauge
+      description: >
+        Percentage of time in kernel space normalized by the number of CPU cores.
+
+    - name: usage.user.ns
+      type: double
+      metric_type: gauge
```
Comment (tetianakravchenko) [inline packages/containerd/data_stream/cpu/fields/fields.yml:29]: unit: nanos
https://github.com/elastic/package-spec/blob/master/versions/1/data_stream/fields/fields.spec.yml#L78

File: packages/containerd/data_stream/blkio/fields/ecs.yml (hunk 2 above)
Comment (MichaelKatsoulis) [reply packages/containerd/data_stream/blkio/fields/ecs.yml:2]: Yes right it is not possible. But I think that container.id is the best place.

File: packages/containerd/data_stream/blkio/fields/ecs.yml (hunk 3 above)
Comment (MichaelKatsoulis) [reply packages/containerd/data_stream/blkio/fields/ecs.yml:1]: It seems that some are added by default like the `ecs.version`, `service.address` and `service.type`. The rest are not default. For example containerd adds `container.id` while kubernetes adds `orchestrator.cluster.name`

**Comment by MichaelKatsoulis** [comment]
@akshay-saraswat could you take a look at the screenshot of the dashboard I have created for containerd? Unfortunately we cannot extract informations from the metrics like containers' statuses which would be something nice to see in a dashboard.

**Review by ChrsMark — COMMENTED** [review]
Overall looks good. Just 2 comments:

1. In the visualisations `Read/Write Bytes Per Container` the Y axis has too big numbers, wondering if we could somehow curate it and make it be shown like 2M instead of 2.000.000 etc. Not sure if possible through Kibana though, maybe there is a setting in the Visualisation options. Also I see that you are using Lens Line charts. I remember in the past we prefered using TSVB charts though for many reasons. @kaiyan-sheng does it happen to have more context here?
2. I see the package has not tests, but you can add them with a mock server already and ensure that package is tested. See https://github.com/elastic/integrations/tree/master/packages/docker/_dev/deploy/docker as an example and a past k8s version https://github.com/elastic/integrations/tree/b189a92444c4dfc33a3462fdad21b40a5bf8ae4e/packages/kubernetes/_dev/deploy/docker

**Comment by MichaelKatsoulis** [comment]
> Overall looks good. Just 2 comments:
> 
> 1. In the visualisations `Read/Write Bytes Per Container` the Y axis has too big numbers, wondering if we could somehow curate it and make it be shown like 2M instead of 2.000.000 etc. Not sure if possible through Kibana though, maybe there is a setting in the Visualisation options. Also I see that you are using Lens Line charts. I remember in the past we prefered using TSVB charts though for many reasons. @kaiyan-sheng does it happen to have more context here?

Yes we can. I had set the value type to default instead of bytes. Also I have created the same dashboard with TSVB as well. So I am good either way.

**Comment by kaiyan-sheng** [comment]
@ChrsMark Yep @MichaelKatsoulis got it, setting the value type to bytes will give 2M instead of 2,000,000 🙂

**Comment by MichaelKatsoulis** [comment]
@ChrsMark , @tetianakravchenko could do a final review? I decided to go with the TSVB dashboard. I also added mock tests.

**Comment by tetianakravchenko** [comment]
@MichaelKatsoulis did you update the screenshot after the latest dashboard changes?

**Comment by MichaelKatsoulis** [comment]
> @MichaelKatsoulis did you update the screenshot after the latest dashboard changes?

Yes the screenshot is correct , just without the `TSVB` word in the dashboard name

**Review by tetianakravchenko — APPROVED** [review]
LGTM!

**Review by ChrsMark — APPROVED** [review]
:shipit:

**Pull request merged**